data/core/commands/core.lua

```diff
@@ -130,7 +131,7 @@ command.add(nil, {
               end
             end
             core.error("Cannot open file %s: %s", text, err)
-          elseif path_stat.type == 'dir' then
+          elseif path_stat and path_stat.type == 'dir' then
```

Comment:
I wonder if we could make the definition of `system.get_file_info` more strict here
https://github.com/lite-xl/lite-xl/blob/e8ecf9c40dc41f457543c799470cc85369a0ccf6/docs/api/system.lua#L242-L249
as we always return a valid table when `err` in not `nil`.

Would LuaLS react correctly then?

Reply:
I did try to define another function definition for get_file_info so that it either returns a valid non-nil file or it returns nil, string`. But LuaLS didn't seem to pick up on it being an either or type of situation.

Reply:
It looks like it might get implemented in their next major version (https://github.com/LuaLS/lua-language-server/discussions/3102).

Until then we either use your solution, or we can do something like
```suggestion
          elseif --[[@cast path_stat -nil]] path_stat.type == 'dir' then
```